### PR TITLE
added button - primary solid and primary outline modes

### PR DESCRIPTION
**Pull request by KatarzynaStefanowiczPagepro**
## Description

- [] TODO write description

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

### CODE REVIEW

File: src/components/atoms/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        <SpinnerLoader appearance={{ size: 'medium' }} />
+      ) : (
+        <button
+          className={clsx(
+            'flex text-md font-medium px-16 py-10 rounded-lg min-h-[2.5rem] transition-all duration-normal',
+            {
+              'bg-blue-400 text-base-white hover:bg-blue-300':
+                mode === 'primarySolid',
+              'bg-gray-300': mode === 'primarySolid' && disabled,
+              // eslint-disable-next-line max-len
```
Comment (krawczykkatarzyna) [inline src/components/atoms/Button/index.tsx:41]: it's not good approach, please use `@apply`

File: src/components/atoms/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import React from 'react';
+import { useIntl } from 'react-intl';
+import clsx from 'clsx';
+import { ButtonProps } from './types';
+import SpinnerLoader from '~shared/Preloaders/SpinnerLoader';
+
+const displayName = 'Button';
+
+const Button = ({
```
Comment (krawczykkatarzyna) [inline src/components/atoms/Button/index.tsx:11]: please use type `PropsWithChildren`

File: src/components/atoms/Button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable react/button-has-type */
+
+import React from 'react';
+import { useIntl } from 'react-intl';
+import clsx from 'clsx';
+import { ButtonProps } from './types';
+import SpinnerLoader from '~shared/Preloaders/SpinnerLoader';
+
+const displayName = 'Button';
```
Comment (krawczykkatarzyna) [inline src/components/atoms/Button/index.tsx:9]: it's not needed

File: src/components/atoms/Button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          {...rest}
+        >
+          {buttonText || children}
+        </button>
+      )}
+    </>
+  );
+};
+
+Button.displayName = displayName;
```
Comment (krawczykkatarzyna) [inline src/components/atoms/Button/index.tsx:62]: it's not needed

File: tailwind.config.js [hunk 5]
```diff
@@ -123,6 +123,10 @@ module.exports = {
           500: '#415A77',
         },
       },
+      spacing: {
+        10: '0.625rem',
```
[the remark this answers is not in the document]
Comment (KatarzynaStefanowiczPagepro) [reply tailwind.config.js]: Fixed

File: tailwind.config.js [hunk 6]
```diff
@@ -123,6 +123,10 @@ module.exports = {
...
+        16: '1rem',
```
[the remark this answers is not in the document]
Comment (KatarzynaStefanowiczPagepro) [reply tailwind.config.js]: Fixed

File: src/components/atoms/Button/index.tsx (hunk 3 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/atoms/Button/index.tsx]: According to the guidelines, we need this
https://hackmd.io/lHcVp96OTBqPCpAnA4ckSQ#DisplayName-and-Component-naming

File: src/components/atoms/Button/index.tsx (hunk 4 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/atoms/Button/index.tsx]: According to the guidelines, we need this
https://hackmd.io/lHcVp96OTBqPCpAnA4ckSQ#DisplayName-and-Component-naming

File: src/components/atoms/Button/index.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable react/button-has-type */
+
```
[the remark this answers is not in the document]
Comment (KatarzynaStefanowiczPagepro) [reply src/components/atoms/Button/index.tsx]: Done

File: src/components/atoms/Button/index.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable react/button-has-type */
```
[the remark this answers is not in the document]
Comment (KatarzynaStefanowiczPagepro) [reply src/components/atoms/Button/index.tsx]: This rule is copied from `src/components/shared/Button/Button.tsx` and fixes this problem:
image

File: src/components/atoms/Button/index.tsx (hunk 2 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/atoms/Button/index.tsx]: Done

File: src/components/shared/Extensions/Button/index.tsx [hunk 9]
```diff
@@ -0,0 +1 @@
+export { default } from './Button';
```
Comment (krawczykkatarzyna) [inline src/components/shared/Extensions/Button/index.tsx:1]: according to doc the file extension should be .ts

Comment (KatarzynaStefanowiczPagepro) [reply src/components/shared/Extensions/Button/index.tsx]: Fixed

**Review by krawczykkatarzyna — APPROVED** [review]

**Pull request merged**